src/backend/distributed/deparser/deparse_database_stmts.c

```diff
@@ -157,54 +195,110 @@ AppendAlterDatabaseStmt(StringInfo buf, AlterDatabaseStmt *stmt)
 }
 
 
+static void
+AppendAlterDatabaseSetTablespace(StringInfo buf, DefElem *def, char *dbname)
+{
+	appendStringInfo(buf,
+					 "ALTER DATABASE %s SET TABLESPACE %s",
+					 quote_identifier(dbname), quote_identifier(defGetString(def)));
+}
+
+
+/*
+ * Appends basic ALTER DATABASE options to a string buffer.
+ * Basic options are those that can be appended to the ALTER DATABASE statement
+ * after the "WITH" keyword.(i.e. ALLOW_CONNECTIONS, CONNECTION LIMIT, IS_TEMPLATE)
+ * The tablespace option is not a basic option since it is defined with SET option.
+ *
+ * This function takes a string buffer, a DefElem representing a database option,
+ * a boolean indicating whether the prefix "ALTER DATABASE <dbname> WITH" has
+ * already been appended, and a database name. It appends the SQL representation
+ * of the database option to the string buffer.
+ *
+ * Returns:
+ *   A boolean indicating whether the prefix "ALTER DATABASE <dbname> WITH" has
+ *   been appended to the buffer. This is the same as the input
+ *   prefixAppendedForBasicOptions if the prefix was already appended, or true
+ *   if this function appended the prefix.
+ */
+static void
+AppendBasicAlterDatabaseOptions(StringInfo buf, DefElem *def, bool *
+								prefixAppendedForBasicOptions, char *dbname)
```

Comment:
can we please get rid of prefixAppendedForBasicOptions and append ALTER DATABASE %s WITH in call-site?

Reply:
Fixed now. Can you check again?